tests/Behat/AuthenticationContext.php

```diff
@@ -105,6 +104,29 @@ public function backupCasConfigs(): void {
     }
   }
 
+  /**
+   * Act after reverting the config.
+   *
+   * Some tests are changing the oe_authentication:redirect_user_register_route
+   * config value. Changing this value, requires router rebuilding. This is
+   * covered by UserRegisterRouteRedirectConfigSubscriber event subscriber. The
+   * subscriber detects if the value of redirect_user_register_route has been
+   * changed and rebuilds the routes. However, because Behat tests are running
+   * in a single request, often the static cache is not invalidated, resulting
+   * in a failure to detect the config value changes, when the configs are
+   * reverted, in ConfigContext::cleanConfig(). Thus, we're manually rebuilding
+   * the routes at the end of the scenario for tests that requesting it, by
+   * using the @RebuildRouter tag.
+   *
+   * @see \Drupal\oe_authentication\Event\UserRegisterRouteRedirectConfigSubscriber
+   * @see \Drupal\DrupalExtension\Context\ConfigContext::cleanConfig()
+   *
+   * @AfterScenario @RebuildRouter
+   */
+  public function rebuildRoutes(): void {
+    \Drupal::service('router.builder')->rebuild();
```

Comment:
I am not sure that this is the right approach. If we are changing a config value in a test and this will persist in the following tests then we are violating the principle that each test should be standalone and not affect any other tests.

In reality, any tests that depend on the default value will start failing once we change this value. Meaning that these tests might run fine the first time, but will fail if we run them a second time.

I think we should remove this section and instead revert the configuration change at the end of the test.